lib/_stream_wrap.js

```diff
@@ -49,7 +50,8 @@ function StreamWrap(stream) {
       this.pause();
       this.removeListener('data', ondata);
 
-      self.emit('error', new Error('Stream has StringDecoder'));
+      self.emit('error', new errors.Error('ERR_ASSERTION',
```

Comment:
`ERR_ASSERTION` is the wrong code to use here. That should only be used with the `assert` module. I recommend creating a new code specific to this condition... perhaps `ERR_STREAM_HAS_STRINGDECODER`